ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleBaseAdapter.java

```diff
@@ -0,0 +1,146 @@
+    String[] splits = versionString.split("\\.");
+
+    if (splits.length >= 3) {
+      int major = Integer.parseInt(splits[0]);
+      int minor = Integer.parseInt(splits[1]);
+      int micro = Integer.parseInt(splits[2]);
+      return new VersionInfo(major, minor, micro);
+    }
+
+    String logMessage =
+        String.format(
+            "Unexpected SDK version format: %s. Returning 0.0.0 for SDK version.", versionString);
+    Log.w(TAG, logMessage);
+    return new VersionInfo(0, 0, 0);
+  }
+
+  @Override
+  public void initialize(
+      @NonNull Context context,
+      @NonNull final InitializationCompleteCallback initializationCompleteCallback,
+      @NonNull List<MediationConfiguration> mediationConfigurations) {
+
+    if (Vungle.isInitialized()) {
+      initializationCompleteCallback.onInitializationSucceeded();
+      return;
+    }
+
+    HashSet<String> appIDs = new HashSet<>();
+    for (MediationConfiguration configuration : mediationConfigurations) {
+      Bundle serverParameters = configuration.getServerParameters();
+      String appIDFromServer = serverParameters.getString(KEY_APP_ID);
+
+      if (!TextUtils.isEmpty(appIDFromServer)) {
+        appIDs.add(appIDFromServer);
+      }
+    }
+
+    int count = appIDs.size();
+    if (count <= 0) {
+      initializationCompleteCallback.onInitializationFailed(
+          "Initialization failed: Missing or Invalid App ID.");
+      return;
+    }
+
+    String appID = appIDs.iterator().next();
+    if (count > 1) {
+      String logMessage =
+          String.format(
+              "Multiple '%s' entries found: %s. Using '%s' to initialize the Vungle SDK.",
+              KEY_APP_ID, appIDs.toString(), appID);
+      Log.w(TAG, logMessage);
+    }
+
+    VungleInitializer.getInstance()
+        .initialize(
+            appID,
+            context.getApplicationContext(),
+            new VungleInitializationListener() {
+              @Override
+              public void onInitializeSuccess() {
+                initializationCompleteCallback.onInitializationSucceeded();
+              }
+
+              @Override
+              public void onInitializeError(String errorMessage) {
+                initializationCompleteCallback.onInitializationFailed(
+                    "Initialization Failed: " + errorMessage);
+              }
+            });
+  }
+
+  @Override
+  public void loadRtbInterstitialAd(
+      @NonNull MediationInterstitialAdConfiguration mediationInterstitialAdConfiguration,
+      @NonNull MediationAdLoadCallback<MediationInterstitialAd, MediationInterstitialAdCallback> mediationAdLoadCallback) {
+    Log.d(TAG, "loadRtbInterstitialAd...");
+  }
+
+  @Override
+  public void loadInterstitialAd(
+      @NonNull MediationInterstitialAdConfiguration mediationInterstitialAdConfiguration,
+      @NonNull MediationAdLoadCallback<MediationInterstitialAd, MediationInterstitialAdCallback> mediationAdLoadCallback) {
+    Log.d(TAG, "loadInterstitialAd...");
```

Comment:
@HuFangshuai Since these are not called, can we add no-op comment here

Reply:
I will remove these two methods, just for tests currently.